lib/form1010_ezr/service.rb

```diff
@@ -120,48 +121,29 @@ def post_fill_required_fields(parsed_form)
       parsed_form.merge!(required_fields)
     end
 
-    # Due to issues with receiving submissions that do not include the Veteran's DOB, we'll
-    # try to add it in before we validate the form
-    def post_fill_veteran_date_of_birth(parsed_form)
-      return if parsed_form['veteranDateOfBirth'].present?
+    # Due to issues with receiving submissions that do not include the Veteran's DOB, full name, SSN, and/or
+    # gender, we'll try to add them in before we validate the form
+    def post_fill_required_user_fields(parsed_form)
+      # User fields that are required in the 10-10EZR schema, but not editable on the frontend
+      required_user_form_fields = {
+        'veteranDateOfBirth' => @user.birth_date,
+        'veteranFullName' => @user.full_name_normalized&.compact&.stringify_keys,
+        'veteranSocialSecurityNumber' => @user.ssn_normalized,
+        'gender' => @user.gender
+      }
 
-      StatsD.increment("#{Form1010Ezr::Service::STATSD_KEY_PREFIX}.missing_date_of_birth")
+      required_user_form_fields.each do |key, value|
+        next if parsed_form[key.to_s].present?
```

Comment:
🤔 Do we need the `to_s` here given the `key`s are defined as strings above?
Alternatively, how would you feel about keeping the `to_s` and changing the keys to symbols for the hash definition shorthand above? (e.g., `veteranDateOfBirth: @user.birth_date,`)

Regardless, this wound up a lot cleaner - nice!

Reply:
I think making the keys symbols would be an issue because the hash is in JSON format. Finding `parsed_form[:veteranDateOfBirth]` would result in a `nil` value.